packages/ODEs/src/constants.ts

```diff
@@ -0,0 +1,369 @@
+// Control constants
+export const CONTROL_TAG = '#';
+export const CONTROL_TAG_LEN = CONTROL_TAG.length;
+export const DF_NAME = 'df';
+const META = `${CONTROL_TAG}meta`;
+export const MAX_LINE_CHART = 6;
+
+/** Control expressions for the problem specifying */
+export enum CONTROL_EXPR {
+    NAME = `${CONTROL_TAG}name`,
+    TAGS = `${CONTROL_TAG}tags`,
+    DESCR = `${CONTROL_TAG}description`,
+    DIF_EQ = `${CONTROL_TAG}equations`,
+    EXPR = `${CONTROL_TAG}expressions`,
+    ARG = `${CONTROL_TAG}argument`,
+    INITS = `${CONTROL_TAG}inits`,
+    CONSTS = `${CONTROL_TAG}constants`,
+    PARAMS = `${CONTROL_TAG}parameters`,
+    TOL = `${CONTROL_TAG}tolerance`,
+    LOOP = `${CONTROL_TAG}loop`,
+    UPDATE = `${CONTROL_TAG}update`,
+    RUN_ON_OPEN = `${META}.runOnOpen`,
+    RUN_ON_INPUT = `${META}.runOnInput`,
+    OUTPUT = `${CONTROL_TAG}output`,
+};
+
+/** Loop consts */
+export enum LOOP {
+  MIN_LINES_COUNT = 1,
+  COUNT_IDX = 0,
+  COUNT_NAME = '_count',  
+  MIN_COUNT = 1,
+};
+
+/** UPDATE consts */
+export enum UPDATE {
+  MIN_LINES_COUNT = 1,
+  DURATION_IDX = 0,
+  DURATION = '_duration',  
+  MIN_DURATION = 0,
+};
+
+/** Basic template illustrating the simplest features */
+const TEMPLATE_BASIC = `${CONTROL_EXPR.NAME}: Template 
+${CONTROL_EXPR.DIF_EQ}:
+  dy/dt = -y + sin(t) / t
+
+${CONTROL_EXPR.ARG}: t
+  initial = 0.01
+  final = 15
+  step = 0.01
+
+${CONTROL_EXPR.INITS}:  
+  y = 0`;
+
+/** Advanced template illustrating extanded features */
+const TEMPLATE_ADVANCED = `NOTES. This is an advanced template. Modify it. Use multi-line formulas if needed.
+Add new equations, expressions, constants & parameters. Edit these notes if required.
+${CONTROL_EXPR.NAME}: Advanced
+${CONTROL_EXPR.DIF_EQ}:
+  dx/dt = E1 * y + sin(t)
+
+  dy/dt = E2 * x - pow(t, 5)
+
+${CONTROL_EXPR.EXPR}:
+  E1 = C1 * exp(-t) + P1
+  E2 = C2 * cos(2 * t) + P2
+
+${CONTROL_EXPR.ARG}: t
+  start = 0
+  finish = 2
+  step = 0.01
+
+${CONTROL_EXPR.INITS}:
+  x = 2
+  y = 0
+
+${CONTROL_EXPR.CONSTS}:
+  C1 = 1
+  C2 = 3
+
+${CONTROL_EXPR.PARAMS}:
+  P1 = 1
+  P2 = -1
+
+${CONTROL_EXPR.TOL}: 0.00005`;
+
+/** Extended template illustrating extanded features */
+const TEMPLATE_EXTENDED = `NOTES. This is an extended template. It has additional scripting annotations.
```

Comment:
I suggest to move templates and use cases to separate files - it's going to be a lot easier to read it this way.

Reply:
Done!